recipes/opencv/all/conanfile.py

```diff
@@ -0,0 +1,214 @@
+        self._cmake.definitions["WITH_CUBLAS"] = False
+        self._cmake.definitions["WITH_NVCUVID"] = False
+        self._cmake.definitions["WITH_FFMPEG"] = False
+        self._cmake.definitions["WITH_GSTREAMER"] = False
+        self._cmake.definitions["WITH_OPENCL"] = False
+        self._cmake.definitions["WITH_CUDA"] = False
+        self._cmake.definitions["WITH_ZLIB"] = True
+
+        self._cmake.definitions["WITH_JPEG"] = self.options.with_jpeg
+        self._cmake.definitions["WITH_PNG"] = self.options.with_png
+        self._cmake.definitions["WITH_TIFF"] = self.options.with_tiff
+        self._cmake.definitions["WITH_JASPER"] = self.options.with_jasper
+        self._cmake.definitions["WITH_OPENEXR"] = self.options.with_openexr
+        self._cmake.definitions["OPENCV_MODULES_PUBLIC"] = "opencv"
+
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.definitions["BUILD_WITH_STATIC_CRT"] = "MT" in str(self.settings.compiler.runtime)
+        self._cmake.configure(build_folder=self._build_subfolder)
+
+        return self._cmake
+
+    def build(self):
+        self._patch_opencv()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "staticlib"))
+        tools.remove_files_by_mask(self.package_folder, "*.cmake")
+
+    def package_info(self):
+        version = self.version.split(".")[:-1]  # last version number is not used
+        version = "".join(version) if self.settings.os == "Windows" else ""
+        debug = "d" if self.settings.build_type == "Debug" and self.settings.compiler == "Visual Studio" else ""
+
+        def get_lib_name(module):
+            return "opencv_%s%s%s" % (module, version, debug)
+
+        def add_components(components):
+            # TODO: OpenCV doesn't use cmake target namespace
+            for component, requires in components.items():
+                self.cpp_info.components[component].names["cmake_find_package"] = "opencv_" + component
+                self.cpp_info.components[component].names["cmake_find_package_multi"] = "opencv_" + component
+                self.cpp_info.components[component].libs = [get_lib_name(component)]
+                self.cpp_info.components[component].requires = requires
+                if self.settings.os == "Linux":
+                    self.cpp_info.components[component].system_libs = ["dl", "m", "pthread", "rt"]
+
+        self.cpp_info.filenames["cmake_find_package"] = "OpenCV"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "OpenCV"
+
+        add_components({
+            "core": ["zlib::zlib"],
```

Comment:
how did you get this?

Reply:
I followed the original cmake targets generated by OpenCV.

Reply:
I advice to prefix conan component names to avoid clashes in pkg-config side.